prov/opx/include/rdma/opx/fi_opx.h

```diff
@@ -50,6 +50,18 @@
 #include <uthash.h>
 #include <ofi_list.h>
 
+#if !defined(OPX_AV)
+#define OPX_AV FI_AV_UNSPEC
+#endif
+
+#if !defined(OPX_THREAD)
+#define OPX_THREAD FI_THREAD_ENDPOINT
+#endif
+
+#if !defined(OPX_MR)
+#define OPX_MR FI_MR_BASIC
+#endif
```

Comment:
Why do these need to be added here? They should already be defined from prov/opx/configure.m4. If they're not, then configure.m4 should be fixed to do so, rather than adding them here.

Reply:
@belynam the build process assumes several of these macro definitions are created, it turns out they are not created. The documentation for OPX provides those as default value. If ya'll would prefer them placed in configure.m4 I'll go ahead and do that.

Reply:
correction, these preprocessor variables supposed to be defined in the configure.ac script using `AC_DEFINE_UNQUOTED` for opx. During compilation of the ops provider, the compiler claims these variables have not defined. Is this a configuration variable caching issue or is there a specific fix for this situation you all would prefer to see?

Reply:
These 3 values (OPX_AV, OPX_THREAD, OPX_MR) are defined in our configure.m4 already. For example, OPX_AV is defined [here](https://github.com/ofiwg/libfabric/blob/2f605226ce5c41106b72a664f711c91dcb7d6ca0/prov/opx/configure.m4#L74).

If they are not getting defined, that would suggest that portion of the configure script is not being executed, even though it should be.